packages/terra-button-group/src/ButtonGroup.jsx

```diff
@@ -60,17 +62,26 @@ class ButtonGroup extends React.Component {
   handleKeyDown(event, idx) {
     const allBtns = this.btnGrpRef.querySelectorAll('[data-terra-button-group-button]');
     let key = idx;
-
-    if (event.keyCode === KEY_RIGHT && allBtns[key + 1]) {
-      key += 1;
-      while (allBtns[key] && allBtns[key].hasAttribute('disabled')) {
+    if ((event.keyCode === KEY_RIGHT || event.keyCode === KEY_DOWN)) {
```

Comment:
we need to include `allBtns[key + 1]` check as well in order to prevent an error of accessing length when there is no data on array `allBtns`

Reply:
We won't get any errors, if it is last element, it'll come to first element focus as we implement Cyclic Behavior.

Reply:
This behavior needs to be showcased through a unit or wdio test.

Reply:
wdio test cases updated.